doc/_extensions/inventory_builder.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This is based on the dummy builder from Sphinx, which is
+# BSD licensed.
+#
+# This package can be added as an extension, and if run as a
+# builder (sphinx -b inventory), it will not generated any
+# output files apart from an inventory file (objects.inv).
+#
+# It can be run as a 1st step when projects contain multi-
+# directional links between them.
+
+from pathlib import Path
+from typing import Iterator, Set
+
+from docutils import nodes
+from sphinx.builders import Builder
+from sphinx.util.inventory import InventoryFile
+
+
+class InventoryBuilder(Builder):
+    name = 'inventory'
+    epilog = 'Only generates an inventory file.'
+
+    allow_parallel = True
+
+    def init(self) -> None:
+        pass
+
+    def get_outdated_docs(self) -> Iterator[str]:
+        for doc_name in self.env.found_docs:
```

Comment:
I didn't review the earlier PRs in detail so I'm just asking for my own knowledge: how does this work?

I don't see 'env' in the Builder API ref here: https://www.sphinx-doc.org/en/master/extdev/builderapi.html Not sure if that is the right place to look, though.

Reply:
`self.env` contains an instance of `BuildEnvironment`, ref. https://www.sphinx-doc.org/en/master/extdev/envapi.html#sphinx.environment.BuildEnvironment. Other builder support this feature (HTML, XML...) see, e.g. https://github.com/sphinx-doc/sphinx/blob/4.x/sphinx/builders/xml.py#L46